### ISSUE TITLE
 Interop Lack of Transparency & Accountability

### ISSUE THREAD

**Issue by mtom55** [web-platform-tests/interop#888]
### Description

The voting process for Interop proposals is done in **secret** and each browser vendor involved can veto any feature without any transparency. Browser Vendors can sift through interop proposals and exclude items they believe are too difficult or don’t match their internal corporate goals. This is bad for the web and against the principle of openness and transparency that web standardization efforts rely on.  

**We propose that any votes or vetoes are made public with attribution, and any reasons for not proceeding with a particular feature is posted in the github issue along with that attribution.  A summary page should be published for each future Interop with the complete set of information.**

This lack of transparency has already been negatively received by developers and has been a source of mistrust and dissatisfaction with the interop process:

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;*"Fans of the spec bemoan lack of transparency in Interop 2024 process"*  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[Thomas Claburn \- The Register](https://www.theregister.com/2024/02/03/jpeg_xl_interop_2024/)

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;*"That's it? That's the response to what is, by several times, the most requested feature in Interop 2024? 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;No explanation for rejecting the feature that got 4.5x more support than the runner-up? Really?"*  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[Tibet Tornaci](https://github.com/web-platform-tests/interop/issues/430#issuecomment-1923216914)

This problem is also discussed in [this issue](https://github.com/web-platform-tests/interop/issues/290) which discusses the lack of transparency of Interop.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;*“My assumption and intention was that the process would be transparent, making per-proposal decisions by consensus, 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;and documenting those decisions in meeting notes or equivalent. I made this assumption because almost everything in 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;web standards and WPT is public by default, and the Interop 2022 positions were made public via GitHub issues.”*  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[*Philip Jägenstedt*](https://github.com/web-platform-tests/interop/issues/290)

There are many reasons why keeping the Interop process for choosing which features to work on a secret is problematic. Here are the key issues:

1. **Lack of Transparency**  

   **Why it matters**: Transparency is fundamental to maintaining trust in any process, especially when it's shaping the future of widely used technologies. If the interop voting process is kept secret, it becomes unclear who is influencing the decisions, how the priorities are set, and why certain features are prioritized over others.  
   
   **Impact**: This lack of visibility can lead to suspicion or the perception that decisions are being made behind closed doors or based on undisclosed factors, such as personal or corporate interests.

2. **Bias or Corporate Influence**

   **Why it matters**: If the voting process is secret, there's a risk that powerful stakeholders could exert disproportionate influence over the feature selection, even if those features don't align with the broader needs of the web community. 
    
   **Impact**: Smaller companies, independent developers, or underrepresented groups might not have their voices heard, which could skew development toward the interests of the most powerful players rather than benefiting the web ecosystem as a whole.

3. **Accountability Issues**  

   **Why it matters**: Keeping votes secret removes accountability. If a particular decision turns out to be unproductive or controversial, it can be difficult to trace back to the individuals or groups responsible for the decision. Without accountability, it’s harder to make improvements to the decision-making process.  
   
   **Impact**: If features that are unpopular or less relevant to most developers are prioritized, it could delay or block the advancement of features that are more universally beneficial. Without accountability, it’s harder to rectify such issues.

4. **Excluding Community Feedback**
  
   **Why it matters**: Public visibility into the voting process allows community members to understand why certain features are chosen and provides them an opportunity to give feedback on the prioritization. Keeping the process secret means that community feedback may be excluded.  
   
   **Impact**: This lack of engagement could result in features that don't align with real-world use cases, or miss critical input from people actually working with the web technologies on a day-to-day basis.

5. **Inability to Audit the Process**

   **Why it matters**: When the voting process is secret, it becomes very difficult to audit how decisions were made. For example, if a particular decision seems to have been made under unclear circumstances, there’s no way to verify if the process was fair and unbiased.

   **Impact**: An opaque process undermines confidence in the decision-making system and can erode the legitimacy of the project or initiative.

6. **Potential for Conflicts of Interest**

   **Why it matters**: Secret voting processes might mask potential conflicts of interest that could influence the decision-making. For example, a particular vendor or organization may be pushing for a feature that benefits their product or service but isn't necessarily the best choice for the web as a whole.

   **Impact**: This could lead to the prioritization of features that serve narrow interests rather than the broader needs of the web ecosystem, possibly undermining the long-term health and sustainability of the web.

7. **Gaming the Process to Maintain High Interop Scores**

   **Why it matters:** In a secret voting system, there’s a risk that browser vendors might strategically avoid committing to difficult or complex features in order to maintain or improve their interop score. This could happen if vendors prioritize "easy wins" or less challenging features that are more likely to pass the interop tests with minimal effort, rather than tackling more complex, nuanced features that could be harder to implement across all browsers.  
   
   **Impact:** By gaming the process, vendors could prioritize features that are less impactful but more easily interoperable, while neglecting or delaying more important innovations that require significant effort or cross-vendor collaboration. This would ultimately result in a less innovative web, where difficult but valuable features are delayed or ignored in favor of easier, less meaningful improvements. Such behavior could also discourage long-term investment in true interoperability and lead to stagnation in web technology development.

### **Conclusion**

Overall, **secrecy in the voting process** undermines many of the principles that made the open web what it is today—**transparency &** **community involvement are critical**. 

A more open voting process ensures that decisions are made in the best interest of the broader developer and user community, helping to ensure that the web evolves in a way that is fair, inclusive, and sustainable.

### Specification

Interop Team

### Additional Signals

https://www.theregister.com/2024/02/03/jpeg_xl_interop_2024/

**Comment by o-t-w** [comment]
The Igalia podcast has a useful episode explaining the process  https://podcasts.apple.com/gb/podcast/igalia-chats/id1533323588?i=1000647686844

**Comment by mtom55** [comment]
@nt1m why have you removed this ticket and [891](https://github.com/web-platform-tests/interop/issues/891) from the list to be discussed for Interop 2025?

**Comment by nt1m** [comment]
This Github project is used purely for practical purposes, it only contains potential developer focus areas scored by WPTs. We copy this Github project into spreadsheets for evaluating developer proposals for focus areas, so it's not great for us having to trim down rows every time we run through the different spreadsheets.

#891 is an investigation (not a focus area), it's in fact one that we're already doing this year (see the Interop 2024 dashboard in the investigations and https://github.com/web-platform-tests/interop-mobile-testing). If you look at the [list of issues](https://github.com/web-platform-tests/interop/labels/investigation-effort-proposal), none of the investigations are automatically put in the Github project.

This issue isn't really a focus area in the classic sense either.

**Comment by mtom55** [comment]
> This Github project is used purely for practical purposes, it only contains potential developer focus areas scored by WPTs. We copy this Github project into spreadsheets for evaluating developer proposals for focus areas, so it's not great for us having to trim down rows every time we run through the different spreadsheets.
> 
> #891 is an investigation (not a focus area), it's in fact one that we're already doing this year (see the Interop 2024 dashboard in the investigations and https://github.com/web-platform-tests/interop-mobile-testing). If you look at the [list of issues](https://github.com/web-platform-tests/interop/labels/investigation-effort-proposal), none of the investigations are automatically put in the Github project.
> 
> This issue isn't really a focus area in the classic sense either.
Fair enough, however, given the significant developer interest in both of these issues, particularly this transparency issue, It's reasonable for the web community to expect them to be discussed in detail as part of Interop 2025 and to see the detailed results of those discussions.

We're asking for the details of each voting round to be made publicly available along with reasons for vetos.

**Comment by bivens-dev** [comment]
The final 2025 interop list was actually so incredibly disheartening when it came out. The spectacularly wide gaps that existed between what developers had asked for and what they got out of the process was genuinely depressing.

I don’t understand how some of it could even be said with a straight face. One of this years “highlights” for example involves a certain browser vendor literally running a find and replace across their code base to address an incompatibility that was resolved in versions 2 and 3 of other vendors from memory.

That undermines the entire concept of what interop is supposed to be about and a huge part of that lays squarely at the heart that it’s a secretive voting process that provides zero benefits to the community it is supposed to serve.
